pkg/kapp/app/recorded_app.go

```diff
@@ -23,8 +26,10 @@ const (
 )
 
 type RecordedApp struct {
-	name   string
-	nsName string
+	name       string
+	fqName     string
+	nsName     string
+	isMigrated bool
```

Comment:
I have been thinking about it a bit... Now that we have a label present in the configmap, do we just want to have a method for recorded app called `isMigrated()` that checks if the label is present or not? As it seems like we are storing the same information at 2 places.

Reply:
I was thinking all the same lines. In an earlier conversation we optimized `Rename` / `Delete` and maybe others with the `isMigrated` bool, which was populated in the `Exists()` function.

`Rename` for example would now need to get the metadata and check the label. It doesn't have access to the configmap prior to the function call. So I don't think (though I could be wrong) we can get rid of the `isMigrated` bool, as it cannot be replaced with a check for the labels in all cases.

Let me know if that doesn't make sense / I am misreading your question.

REF: https://github.com/vmware-tanzu/carvel-kapp/pull/403/files/631b31756e11cafc3d1bcd01b44901f69cc08aa5#diff-c4cadf071a66c0e934e58c35eb0a9116fee4c744abd14621dc1eeebe7f5e9fb8R267-R271

Reply:
> Rename for example would now need to get the metadata and check the label

True, and I think the same would need to happen for `Delete`. So `Exists()` might be another place where might want to save the meta data and we should be able to reduce one call. (We can have a seperate PR if we decide to do it).

Reply:
> do we just want to have a method for recorded app called isMigrated() that checks if the label is present or not? As it seems like we are storing the same information at 2 places.

Now I am having similar thoughts about `fqName` as well xD
(Maybe I am just overthinking)